subsys/net/lib/openthread/platform/radio.c

```diff
@@ -361,12 +443,49 @@ otRadioFrame *otPlatRadioGetTransmitBuffer(otInstance *aInstance)
 	return &sTransmitFrame;
 }
 
+static void get_rssi_energy_detected(struct device *dev, s16_t max_ed)
+{
+	ARG_UNUSED(dev);
+	radio_signal_set(RADIO_SIGNAL_BLOCKING, max_ed);
+}
+
 int8_t otPlatRadioGetRssi(otInstance *aInstance)
 {
-	ARG_UNUSED(aInstance);
+	s8_t ret_rssi = INT8_MAX;
+	int signal_value;
+
+	otRadioCaps radioCaps =  otPlatRadioGetCaps(aInstance);
```

Comment:
You should use Zephyr radio API directly here `radio_api->get_capabilities(radio_dev)` and IEEE802154_HW_ENERGY_SCAN.